### ISSUE TITLE
Add descriptions to labels

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#4540]
We've got a [set of conventions](https://stylelint.io/developer-guide/issues) for managing issues. I've retroactively applied these to the open issues that I've come across while combing through the last 3 months of (fantastic) work on the project.

I think it would be beneficial for new contributors to add descriptions to our labels as the distinction between labels can be quiet subtle e.g. "meta" vs "infrastructure". However, it will make the drop-down list longer but I think the benefits outweigh this.

---

We could possibly do with a new welcome doc that directs new contributors to these developer guide resources too.

**Comment by jeddy3** [comment]
Oh my, I got the label conventions wrong on this issue. How ironic. It looks like I need to take another look at that doc myself :)

**Comment by jeddy3** [comment]
Something along the lines of:

- `status: discussion` - needs further discussion by contributors to triage 
- `status: needs clarification` - needs clarification from the author to triage 
- `status: needs investigation` - needs investigation by contributors to triage 
- `status: needs reproducible example` - needs a reproducible repo or demo by the author to triage 
- `status: wip` - is being worked on by someone
- `status: blocked` - is blocked by another issue or pull request

And:

- `pr: needs revision`: pull request needs revision by the author
- `pr: blocked`: pull request is blocked by another issue or pull request
- `pr: needs documentation`: pull request needs documentation adding by the author
- `pr: needs tests`: pull request needs tests adding by the author
- `pr: dependencies`: pull request relates to dependencies

And:

- `syntax: html` - relates to HTML style tags and attributes
- `syntax: scss` - relates to SCSS and SCSS-like syntax
- `syntax: less` - relates to Less syntax

And:

- `priority: high` - is impactful on users

And:

- `type: documentation` - an improvement to the documentation
- `type: enhancement` - a new feature that isn't a rule, option or autofix
- `type: new rule` - a new rule
- `type: new option` - a new option for an existing rule
- `type: new autofix` - autofix support for an existing rule
- `type: infrastucture` - an improvement to the developer operations like the CI
- `type: meta` - an improvement to collaboration platform like GitHub issue
- `type: tests` - an improvement to testing

---

Incidentally, the distinction between `meta` and `infrastructure` is, perhaps, too fine-grained. `meta` is a bit of an ambiguous word too. I'm happy to see `meta` merged into `infrastructure`.

**Comment by hudochenkov** [comment]
Great descriptions!

Maybe we can remove `pull request` in the beginning of all `pr: *` labels? Or replace with “PR” to be more concise.

`pr: dependencies` — do we need this label? I can't come up with single use case I would want to find PR with this label.

Suggestion:

> `priority: high` - breakes core user experice. Doesn't include false positives/negatives.

At least this was my idea to mark really [critical issues](https://github.com/stylelint/stylelint/issues?q=is%3Aopen+is%3Aissue+label%3A%22priority%3A+high%22), when I added this label recently.

---

I would reconsider how we're using `help wanted` label. We add it for every issue, because help is always wanted. There is no help from this label in my opinion. I would remove it completely, or put on only very hard issues, where core contributors lack expertise or something else.

---

We can remove `meta`, I think.

**Comment by jeddy3** [comment]
> Maybe we can remove pull request in the beginning of all pr: * labels? Or replace with “PR” to be more concise.

I thought about that, but it was intentional for new users who might not know the "PR" acronym. 

> pr: dependencies — do we need this label? I can't come up with single use case I would want to find PR with this label.

Me neither. However, I'm not if sure dependabot is the same as greenkeeper, but the latter required a label to be added to its pull requests (I think to help identify fake PRs from users pretending to be greenkeeper).

> priority: high - breakes core user experice. Doesn't include false positives/negatives.

Interesting. However, there might be a false positive bug in a popular rule that impacts a lot of users. I used "is impactful on users" to cover both bugs that impact a few users a lot and those that impact a lot of users a little.

> I would reconsider how we're using `help wanted` label. 

Agreed.

>  I would remove it completely, or put on only very hard issues, where core contributors lack expertise or something else.

I don't think we'll want to remove it as [other tools](http://github-help-wanted.com/) feed from it. 

I feel like there is an overlap between "good first issue" and "help wanted". We (me especially I admit) use them together a lot, whereas I think your excellent suggestion of reserving "help wanted" for hard issues releases "good first issue" for all the bitty issues that we need a hand on. 

How's about:

`good first issue` - is good for newcomers
`help wanted` - is a difficult issue and help is wanted to tackle it

And I'll update [the conventions](https://stylelint.io/developer-guide/issues) to say you can use either of these labels but not both at the same time.

> We can remove `meta`, I think.

Good stuff.

**Comment by hudochenkov** [comment]
> I thought about that, but it was intentional for new users who might not know the "PR" acronym.

We don't have collaborators who don't know what PR is :)

> good first issue - is good for newcomers
> help wanted - is difficult and help is wanted to tackle it

👍

**Comment by jeddy3** [comment]
> We don't have collaborators who don't know what PR is :)

A first-time contributor can submit a pull request via GitHub without ever seeing the acronym. All the buttons and tabs on GitHub use "Pull request". We then add a label with "pr" on it and they might be left wondering what it means, so they hover over the label to see. I'm not precious about keeping the "pull request" bit in the descriptions but I'm keen to do all we can to welcome new contributors who's pull request to stylelint might be their first one ever :)

(I was working with a lot of junior developers last year and it was an eye-opening experience in regards to my own presumed knowledge!) 

>> good first issue - is good for newcomers
>>help wanted - is difficult and help is wanted to tackle it

> 👍

I was thinking about this a little more. I think we also need a label for when an issue is triaged and needs working on i.e. discussions, investigations etc are over, but the issue is neither blocked or wip. Currently "help wanted" is used for this, but I think a more specific `status: needs claiming` label would work better so we can free up "help wanted" for the difficult issues.

We would add it to all the issues that gather comments like "This issue still open a year later...". Which would open us up to using encouraging stock replies like "Thanks for your interest in the issue. It is labeled as "status: needs claiming". If you would like to work on this issue please say so".

**Comment by hudochenkov** [comment]
> status: needs claiming

What about `status: ready for pickup` or `status: ready to implement`? I don't understand your proposal without extra description. Maybe because I'm non-native English speaker.

**Comment by jeddy3** [comment]
Oh, that's nice. I like the use of "ready" as it differentiates it from all the triaging labels, which start with "needs ...":

I think I prefer `status: ready to implement` over  `status: ready for pickup`. Let's start there. Which give us:

- `status: needs discussion` - needs further discussion by contributors to triage 
- `status: needs clarification` - needs clarification from the author to triage 
- `status: needs investigation` - needs investigation by contributors to triage 
- `status: needs reproducible example` - needs a reproducible repo or demo by the author to triage

And: 
 
- `status: ready to implement` - is ready to be worked on by someone
- `status: wip` - is being worked on by someone
- `status: blocked` - is blocked by another issue or pull request

Does anyone else have anything else to add or shall I make these changes? They won't be set in stone and we can improve on the labels going forward. I think having descriptions is a good step forward, though

**Comment by hudochenkov** [comment]
Sounds good!

**Comment by ntwb** [comment]
Sounds good to me, can I suggest just do these for the main repo

We can then sync them from the https://github.com/stylelint/.github/tree/master/.github repo with a GitHub action to all the repos with specific labels for repos as required

**Comment by jeddy3** [comment]
> We can then sync them from the https://github.com/stylelint/.github/tree/master/.github repo with a GitHub action to all the repos with specific labels for repos as required

Wow, that's nice. Do you mean using something [like this](https://github.com/micnncim/action-label-syncer)?

As you said, I'll just make the changes in this repo for now.

**Comment by ntwb** [comment]
Yup, there’s quite a few options, GH Actions has matured so much in recent months

Cheers,

Stephen Edgar
________________________________

**Comment by jeddy3** [comment]
[Done](https://github.com/stylelint/stylelint/labels?sort=name-asc). I had to tweak some of the descriptions as there's less space to work with than I thought.

**Comment by vankop** [comment]
May I suggest to add labels related to upstream `postcss-*` package problem? looks like (unfortunately) this is a common issue.

**Comment by jeddy3** [comment]
> May I suggest to add labels related to upstream postcss-* package problem?

Something more than `syntax: x` + `status: blocked` ([example](https://github.com/stylelint/stylelint/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+blocked%22+label%3A%22syntax%3A+css-in-js%22))?

I'm not opposed to this as I think it'll provide clarity, but I think we'll need to tweak our use of `status: blocked`to make it work. 

We could add a new `upstream: x` class of labels e.g.

- `upstream: postcss-cssinjs`
- `upstream: postcss-html`
- `upstream: postcss-sass`

We'd then retire the use of `status: blocked` for these issues, reserving it for when our own issues or prs are blocking each other.

**Comment by jeddy3** [comment]
Scratch that... they'll be too much overlap with the `syntax:` class of labels.  I propose we simply have an `upstream` label (or something similar). We can then have `syntax: x` + `upstream` combinations.

**Comment by jeddy3** [comment]
I've added an `upstream` label. 

What are the next steps with this issue?  I struggled to find a GitHub action that will read a label.yml from another repo. Some labels are specific to the stylelint repo, e.g. `syntax: *` and `type: new rules` etc. I'm happy to manually run a script a few times to sync the standard labels across our repositories and then close this issue if everyone is satisfied with the labels, descriptions and colours here.

**Comment by ntwb** [comment]
Hmmmm, same, I was sure one existed to do just such a thing

I found this issue requesting support for this to be added
> https://github.com/crazy-max/ghaction-github-labeler/issues/3

Will take another look over the weekend

**Comment by jeddy3** [comment]
Something will hopefully turn up, but it looks like a non-starter.

https://github.com/crazy-max/ghaction-github-labeler/issues/9#issuecomment-565553035:

> But since the file would no longer be in the repository, there would be no way to trigger the build. This is therefore not feasible until GitHub has introduced the addition of custom triggers or manual build triggering.
